src/julia-syntax.scm

```diff
@@ -101,20 +101,31 @@
                                  tuples))))
             `(call (top lastindex) ,a ,dimno))))
 
-;; replace `end` for the closest ref expression, so doesn't go inside nested refs
-(define (replace-end ex a n tuples last)
+(define (begin-val a n tuples last)
+  (if (null? tuples)
+      (if (and last (= n 1))
+          `(call (top firstindex) ,a)
+          `(call (top first) (call (top axes) ,a ,n)))
```

Comment:
I missed this, but it's a bit of a shame that this isn't `firstindex(a, n)` to mirror `lastindex(a, n)`.

Reply:
Sounds like a good non-breaking PR?